holidays/financial/national_stock_exchange_of_india.py

```diff
@@ -64,57 +91,68 @@ def _populate_public_holidays(self):
         # Good Friday.
         self._add_good_friday(tr("Good Friday"))
 
-        # Dr. Baba Saheb Ambedkar Jayanti.
+        # Dr. B. R. Ambedkar Jayanti.
         self._move_holiday(self._add_holiday_apr_14(tr("Dr. Baba Saheb Ambedkar Jayanti")))
 
-        # Maharashtra Day.
-        self._move_holiday(self._add_holiday_may_1(tr("Maharashtra Day")))
+        self._move_holiday(
+            self._add_holiday_may_1(
+                # May Day.
+                tr("May Day")
+                if 2010 <= self._year <= 2014
+                # Maharashtra Day.
+                else tr("Maharashtra Day")
+            )
+        )
 
         # Independence Day.
         self._move_holiday(self._add_holiday_aug_15(tr("Independence Day")))
 
-        # Mahatma Gandhi Jayanti.
+        # Gandhi Jayanti.
         self._move_holiday(self._add_holiday_oct_2(tr("Mahatma Gandhi Jayanti")))
 
         # Christmas Day.
         self._move_holiday(self._add_christmas_day(tr("Christmas Day")))
 
         # Hindu Calendar Holidays.
 
-        # Maha Shivaratri.
-        self._move_holiday(self._add_maha_shivaratri(tr("Maha Shivaratri")))
+        if self._year >= 2007:
+            # Maha Shivaratri.
+            self._move_holiday(self._add_maha_shivaratri(tr("Maha Shivaratri")))
 
         # Holi.
         self._move_holiday(self._add_holi(tr("Holi")))
 
-        # Ram Navami.
-        self._move_holiday(self._add_ram_navami(tr("Ram Navami")))
+        if self._year >= 2006:
+            # Ram Navami.
+            self._move_holiday(self._add_ram_navami(tr("Ram Navami")))
 
-        # Mahavir Jayanti.
-        self._move_holiday(self._add_mahavir_jayanti(tr("Mahavir Jayanti")))
+            # Mahavir Jayanti.
+            self._move_holiday(self._add_mahavir_jayanti(tr("Mahavir Jayanti")))
 
         # Ganesh Chaturthi.
         self._move_holiday(self._add_ganesh_chaturthi(tr("Ganesh Chaturthi")))
 
         # Dussehra.
         self._move_holiday(self._add_dussehra(tr("Dussehra")))
 
-        # Diwali Laxmi Pujan.
-        self._move_holiday(self._add_gau_krida(tr("Diwali Laxmi Pujan")))
+        # Diwali Lakshmi Puja.
+        self._move_holiday(self._add_diwali_india(tr("Diwali Laxmi Pujan")))
 
-        # Diwali Balipratipada.
-        name = tr("Diwali Balipratipada")
-        # NSE's calendar adds an extra-day gap in 2022–2023
-        # between Diwali Laxmi Pujan and Diwali Balipratipada.
-        self._move_holiday(
-            self._add_bhai_dooj(name)
-            if self._year in {2022, 2023}
-            else self._add_govardhan_puja(name)
-        )
+        if self._year <= 2002 or self._year >= 2011:
+            # Diwali Balipratipada.
+            self._move_holiday(self._add_govardhan_puja(tr("Diwali Balipratipada")))
 
         # Guru Nanak Jayanti.
         self._move_holiday(self._add_guru_nanak_jayanti(tr("Guru Nanak Jayanti")))
 
+        if 2003 <= self._year <= 2010:
+            # Bhai Bhij.
```

Comment:
```suggestion
            # Bhai Dooj.
```
en_US common name for this holiday. Also en_US translation needs to be changed.